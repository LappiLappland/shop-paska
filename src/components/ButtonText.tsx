import { ReactNode, useRef } from 'react';
import useRipple from '../hooks/useRipple';

interface ButtonProps {
  children?: ReactNode;
  className?: string;
  color?: 'primary' | 'secondary' | 'tertiary';
  rippleClassName?: string;
  type?: 'submit' | 'reset' | 'button';
  disabled?: boolean;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

export default function ButtonText({
  children,
  className = '',
  color = 'primary',
  rippleClassName = '',
  type,
  disabled,
  onClick,
}: ButtonProps) {
  const buttonRef = useRef<HTMLButtonElement>(null);

  function getColors() {
    switch (color) {
      case 'tertiary':
        return `
        text-tertiary fill-tertiary stroke-tertiary
        hover:bg-tertiary/hovered-r
        focus-visible:bg-tertiary/focused-r
        active:bg-tertiary/pressed-r
        `;
      case 'secondary':
        return `
        text-secondary fill-secondary stroke-secondary
        hover:bg-secondary/hovered-r
        focus-visible:bg-secondary/focused-r
        active:bg-secondary/pressed-r
        `;
      default:
        return `
        text-primary fill-primary stroke-primary
        hover:bg-primary/hovered-r
        focus-visible:bg-primary/focused-r
        active:bg-primary/pressed-r
        `;
    }
  }

  const ripple = useRipple(buttonRef, {
    color: 'bg-primary/focused',
    className: rippleClassName,
  });

  return (
    <button
      ref={buttonRef}
      className={`
      relative flex h-10 items-center justify-center
      px-3
      transition-colors
      ${getColors()}
      bg-transparent
      disabled:fill-on-surface disabled:stroke-on-surface disabled:text-on-surface
      ${className}`}
      type={type}
      disabled={disabled}
      onClick={(e) => {
        if (onClick) onClick(e);
      }}
    >
      {children}
      {ripple}
    </button>
  );
}
