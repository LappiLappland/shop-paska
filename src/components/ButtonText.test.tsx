import { act, render } from '@testing-library/react';
import ButtonText from './ButtonText';

describe('Button component', () => {
  test('renders button with default primary color', () => {
    const { container } = render(<ButtonText>Click me</ButtonText>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders button with secondary color', () => {
    const { container } = render(<ButtonText color="secondary">Click me</ButtonText>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders button with tertiary color', () => {
    const { container } = render(<ButtonText color="tertiary">Click me</ButtonText>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders disabled button', () => {
    const { container } = render(<ButtonText disabled>Click me</ButtonText>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onClick handler when button is clicked', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<ButtonText onClick={onClick}>Click me</ButtonText>);
    const button = getByRole('button');
    act(() => {
      button.click();
    })
    expect(onClick).toHaveBeenCalled();
  });
});