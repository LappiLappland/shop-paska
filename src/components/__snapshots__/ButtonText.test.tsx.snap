// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button with default primary color 1`] = `
<button
  class="
      relative flex h-10 items-center justify-center
      px-3
      transition-colors
      
        text-primary fill-primary stroke-primary
        hover:bg-primary/hovered-r
        focus-visible:bg-primary/focused-r
        active:bg-primary/pressed-r
        
      bg-transparent
      disabled:fill-on-surface disabled:stroke-on-surface disabled:text-on-surface
      "
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;

exports[`Button component renders button with secondary color 1`] = `
<button
  class="
      relative flex h-10 items-center justify-center
      px-3
      transition-colors
      
        text-secondary fill-secondary stroke-secondary
        hover:bg-secondary/hovered-r
        focus-visible:bg-secondary/focused-r
        active:bg-secondary/pressed-r
        
      bg-transparent
      disabled:fill-on-surface disabled:stroke-on-surface disabled:text-on-surface
      "
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;

exports[`Button component renders button with tertiary color 1`] = `
<button
  class="
      relative flex h-10 items-center justify-center
      px-3
      transition-colors
      
        text-tertiary fill-tertiary stroke-tertiary
        hover:bg-tertiary/hovered-r
        focus-visible:bg-tertiary/focused-r
        active:bg-tertiary/pressed-r
        
      bg-transparent
      disabled:fill-on-surface disabled:stroke-on-surface disabled:text-on-surface
      "
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;

exports[`Button component renders disabled button 1`] = `
<button
  class="
      relative flex h-10 items-center justify-center
      px-3
      transition-colors
      
        text-primary fill-primary stroke-primary
        hover:bg-primary/hovered-r
        focus-visible:bg-primary/focused-r
        active:bg-primary/pressed-r
        
      bg-transparent
      disabled:fill-on-surface disabled:stroke-on-surface disabled:text-on-surface
      "
  disabled=""
>
  Click me
  <span
    class="absolute inset-0 h-full w-full overflow-hidden "
  />
</button>
`;
